### PR TITLE
log-conventions

### PR DESCRIPTION
**Pull request by mkawalek**
* logs stick to log-conventions
* connector use firehose

### CODE REVIEW

File: server/lib/lead/get-lead-ident.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ export default function getLeadsIdent(ctx: Object, lead: Object): Object {
   const ident = {};
 
   ident.anonymous_id = `intercom:${lead.user_id}`;
+  ident.external_id = `intercom:${lead.user_id}`;
```
Comment (mkawalek) [inline server/lib/lead/get-lead-ident.js]: I am not sure if this is right. @michaloo could you please take a look at it ? :)

File: server/jobs/handle-bulk.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ export default function handleBulk(ctx, payload) {
         })()
           .then(() => {
             users.map(u => {
-              return ctx.client.logger.info("outgoing.user.success", _.pick(u, ["email", "id"]));
+              return ctx.client.logger.info("outgoing.user.success", _.pick(u, ["email", "hull_id", "external_id"]));
```
Comment (michaloo) [inline server/jobs/handle-bulk.js]: `user` object doesn't have the `hull_id`, it only has `id` which should be mapped to the `hull_id`.

File: server/lib/lead/get-lead-ident.js (hunk 1 above)
Comment (michaloo) [inline server/lib/lead/get-lead-ident.js]: no, we can't modify the `ident` here. For `incoming` logs we use only the ident information we have. Don't worry if we are missing something.
The ident params here affects the way user is created and updated on the platform side.

File: server/lib/lead/post-leads.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ export default function postLeads(ctx: Object, leads: Array<Object>): Promise {
     return service.intercomClient.post("/contacts")
       .send(lead)
       .then(response => {
-        client.logger.info("outgoing.user.success", lead);
+        client.logger.info("outgoing.user.success", { hull_id: lead.id, external_id: lead.user_id, email: lead.email });
```
Comment (michaloo) [inline server/lib/lead/post-leads.js]: Unfortunately we need to move that log line above in call stack to a place where `postLeads` is called.
the reason is that the `lead` object here is not a hull user object, but it was already mapped and converted to object which is sent to Intercom API.

File: server/jobs/send-leads.js [hunk 4]
```diff
@@ -28,7 +31,7 @@ export default function sendLeads(ctx, payload) {
         u["traits_intercom/id"] = intercomData.id;
         u["traits_intercom/tags"] = intercomData.tags.tags.map(t => t.name);
 
-        ctx.client.logger.info("outgoing.user.success", _.pick(u, ["email", "id", "external_id"]));
+        ctx.client.logger.info("outgoing.user.success", _.pick(u, ["email", "hull_id", "external_id"]));
```
Comment (michaloo) [inline server/jobs/send-leads.js]: same story - no `hull_id` in the object.

File: server/jobs/handle-batch.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ function batchHandler(ctx, source, segmentId) {
       return ctx.service.syncAgent.updateUserSegments(u, { add_segment_ids: [segmentId] }, ignoreFilter);
     }));
 
-    users.map(u => ctx.client.logger.debug("outgoing.user.start", _.pick(u, ["email", "id"])));
+    users.map(u => ctx.client.logger.debug("outgoing.user.start", { email: u.email, hull_id: u.id }));
```
Comment (michaloo) [inline server/jobs/handle-batch.js]: `asUser`

File: server/jobs/save-events.js [hunk 6]
```diff
@@ -37,12 +37,16 @@ export default function saveEvents(ctx, payload) {
...
+          const asUser = client.asUser(ident);
+          asUser.logger.info("incoming.event.skip", { reason: "skipping tag event", props, context, eventName });
+          return asUser.traits(traits);
         }
 
-        client.logger.info("incoming.event", { ident, eventName, props, context });
         metric.increment("ship.incoming.events", 1);
-        return client.asUser(ident).track(eventName, props, context);
+        const asUser = client.asUser(ident);
+        asUser.logger.info("incoming.event", { eventName, props, context });
```
Comment (michaloo) [inline server/jobs/save-events.js]: lets split that line into `success` and `error` in correct promise handling

File: server/jobs/save-leads.js [hunk 7]
```diff
@@ -41,9 +41,8 @@ export default function saveLeads(ctx: Object, payload: Object): Promise<String>
     traits["intercom/is_lead"] = true;
     traits["intercom/lead_user_id"] = lead.user_id;
 
-    ctx.client.logger.info("incoming.user.success", { ident, traits });
-    return client
-      .asUser(ident, { active: true })
-      .traits(traits);
+    const asUser = client.asUser(ident, { active: true });
+    asUser.logger.info("incoming.user.success", { traits });
```
Comment (michaloo) [inline server/jobs/save-leads.js]: lets move it into the promise handlers and add `error` log line

File: server/jobs/send-leads.js [hunk 8]
```diff
@@ -12,13 +12,16 @@ export default function sendLeads(ctx, payload) {
 
   ctx.client.logger.debug("sendLeads.preFilter", leads.length);
   const leadsToSave = leads.filter(lead => !syncAgent.userWithError(lead));
+  leadsToSave.map(u => ctx.client.logger.debug("outgoing.user.start", { hull_id: u.id, external_id: u.user_id, email: u.email }));
```
Comment (michaloo) [inline server/jobs/send-leads.js]: `asUser`

File: README.md [hunk 9]
```diff
@@ -63,6 +63,17 @@ docker-compose run npm install dep -s
 
 ---
 
-## Custom logging messages
-
-  - `incoming.event.tagNotFound` - when capturing (un)tag events we update the `intercom/tags` trait - but sometimes we cannot find corresponding tag in `/tags` results
+### Logs 
+
+  Logs that are specific for Intercom Connector: 
+
```
Comment (mkawalek) [inline README.md]: @sbellity sure :)

File: server/jobs/handle-batch.js [hunk 10]
```diff
@@ -12,7 +12,7 @@ function batchHandler(ctx, source, segmentId) {
       return ctx.service.syncAgent.updateUserSegments(u, { add_segment_ids: [segmentId] }, ignoreFilter);
     }));
 
-    users.map(u => ctx.client.logger.debug("outgoing.user.start", _.pick(u, ["email", "id"])));
+    users.map(u => ctx.client.asUser(_.pick(u, ["email", "id"])).logger.debug("outgoing.user.start"));
```
Comment (michaloo) [inline server/jobs/handle-batch.js]: @mkawalek lets add the whole payload here

**Pull request merged**